webhooks-extension/src/components/WebhookBranches/WebhookBranches.js

```diff
@@ -0,0 +1,151 @@
+import React, { Component } from "react";
+import { Modal } from "carbon-components-react";
+import { fetchBranches } from "../../api";
+
+import {
+  DataTable,
+  DataTableSkeleton,
+  InlineNotification
+} from "carbon-components-react";
+
+const {
+  TableContainer,
+  Table,
+  TableHead,
+  TableRow,
+  TableBody,
+  TableCell,
+  TableHeader
+} = DataTable;
+
+export class WebhookBranches extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      rows: [],
+      loading: false,
+      error: null
+    };
+  }
+
+  componentDidMount() {
+    this.setState({
+      loading: true
+    });
+    let [, user, repo] = this.props.webhook
+      .toLowerCase()
+      .replace(/https?:\/\//, "")
+      .split("/");
+
+    fetchBranches(user, repo)
+      .then(branches => {
+        this.setState({
+          rows: branches.map(branch => ({
+            id: `${branch.name}-branch`,
+            branch: branch.name,
+            time: "No Results Available",
+            status: "No Results Available"
+          })),
+          loading: false
+        });
+      })
+      .catch(error => {
+        error.response.text().then(text => {
+          this.setState({
+            error: text,
+            rows: [
```

Comment:
Do we need to include this dummy row in the error state? Can we just leave rows empty and display the error?

Reply:
ill see if it breaks without it, if not then ill remove it